test/e2e/logger.go

```diff
@@ -93,6 +98,19 @@ func NewPausableLogger(opts ...LoggerOption) PausableLogger {
 	}
 }
 
+// WithSwitchWriter returns a LoggerOption that sets the output writer to the given SwitchWriter.
+func WithSwitchWriter(w *SwitchWriter) LoggerOption {
+	return &withSwitchWriterOption{writer: w}
+}
+
+type withSwitchWriterOption struct {
+	writer *SwitchWriter
```

Comment:
does this need to be a SwitchWriter?
or can it be just a writer
```suggestion
	writer io.Writer
```